app/controllers/api/alert_definition_profiles_controller.rb

```diff
@@ -21,5 +21,68 @@ def edit_resource(type, id = nil, data = {})
         raise BadRequestError, "Failed to update alert definition profile - #{err}"
       end
     end
+
+    def get_assignments_resource(_type, id, _data)
+      profile = MiqAlertSet.find(id)
+      profile.get_assigned_tos
+    end
+
+    def get_resource_from_href(href)
+      hrefobj = Href.new(href)
+      resource_search(hrefobj.subject_id, hrefobj.subject, collection_class(hrefobj.subject))
+    end
+
+    def assign_resource(_type, id = nil, data = {})
+      profile = MiqAlertSet.find(id)
+      raise BadRequestError, "Must specify either objects or tags for the assignment target" unless data.include?("objects") || data.include?("tags")
+      if data.include?("objects")
+        objects = data["objects"].collect do |ref|
+          get_resource_from_href(ref)
+        end
+        profile.assign_to_objects(objects)
+      end
+
+      if data.include?("tags")
+        data["tags"].each do |tag|
+          profile.assign_to_tags([get_resource_from_href(tag["href"]).classification], tag["class"])
+        end
+      end
+      profile.get_assigned_tos
+    end
+
+    def unassign_resource(_type, id = nil, data = {})
+      profile = MiqAlertSet.find(id)
+      raise BadRequestError, "Must specify either objects or tags for the assignment target" unless data.include?("objects") || data.include?("tags")
+      # the model does not have a method to remove a specific assignment
+      # so we save the current assignments, remove all assignments,
+      # and then re-add them without the one we wanted to remove
```

Comment:
In regards to this comment, have you considered adding a method on the model to perform this logic? Would be nice to keep this logic out of the controller - and would probably be easier to test as well.